### DISCUSSION TITLE
difference between building in eclipse UI and CLI

### DISCUSSION THREAD

**Discussion by elsazac** [eclipse-platform/eclipse.platform.ui#1473] [Q&A] [answered]
I observed that building in the command line (mvn) takes considerably huge amount of time(with lot of dependency downloads etc.), even comparing with project->clean build.Are the two builds doing exactly the same?

**Comment by laeubi** [comment]
It depends on what "commandline" you compare ... beside that "UI" builds are never the same as "commandline" as they are using different tools/concept.

**Reply by elsazac** [reply]
by command line I meant this one : `mvn clean verify -Pbuild-individual-bundles`

**Reply by laeubi** [reply]
> by command line I meant this one : `mvn clean verify -Pbuild-individual-bundles`

This for example executes all tests what is not happening within eclipse, by the `mvn clean -T1C -DskipTests` should be much faster (if you just want to compile everything).

**Comment by mickaelistria** [comment, marked as answer]
"Project > Clean Build" does use Eclipse JDT incremental builder and doesn't invoke `mvn` at all even for Maven projects. What happens is that Eclipse has an internal translation of project configuration (settings, dependencies) according to Maven (Eclipse m2e provides that) and this configuration is reused for a clean build, so only .class files are generated accordingly.
The `mvn` command-line does resolution of dependencies, check for remote artifacts, extra configuration and many more things more than "just" compiling, as a result the operation is much longer.
The only drawback with this approach is that it can happen for some cases (depending on the Maven plugins being used or other Maven settings that are not supported well by Eclipse m2e or JDT) that the internal translation of Maven settings into Eclipse JDT is incorrect and that the Eclipse build generates output that is not consistent with Maven settings. But those case are not too frequent.

**Comment by elsazac** [comment]
Thanks @laeubi and @mickaelistria for the response and detailed explanation.I understand the difference now.
